src/components/Common/RoleSelect.tsx

```diff
@@ -0,0 +1,61 @@
+import { useQuery } from "@tanstack/react-query";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import Autocomplete from "@/components/ui/autocomplete";
+
+import query from "@/Utils/request/query";
+import { mergeAutocompleteOptions } from "@/Utils/utils";
+import { Role } from "@/types/emr/role/role";
+import roleApi from "@/types/emr/role/roleApi";
+
+interface RoleSelectProps {
+  value?: Role;
+  onChange: (value: Role) => void;
+  disabled?: boolean;
+  className?: string;
+}
+
+export function RoleSelect({
+  value,
+  onChange,
+  disabled,
+  className,
+}: RoleSelectProps) {
+  const { t } = useTranslation();
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const { data: roles, isLoading } = useQuery({
+    queryKey: ["roles", searchTerm],
+    queryFn: query.debounced(roleApi.listRoles, {
+      queryParams: { name: searchTerm },
+    }),
+  });
+
+  const options = mergeAutocompleteOptions(
+    roles?.results.map((role) => ({
+      label: role.name,
+      value: role.id,
+    })) || [],
+    value ? { label: value.name, value: value.id } : undefined,
+  );
+
+  return (
+    <Autocomplete
```

Comment:
Hmm, we need to display descriptions as well, so

- modify autocomplete to accept a prop to support that
- copy over similar functionality
- revert back to using select/etc and integrate infinite query